be/src/exprs/expr_context.cpp

```diff
@@ -191,7 +191,6 @@ StatusOr<ColumnPtr> ExprContext::evaluate(Expr* e, vectorized::Chunk* chunk) {
 #ifndef NDEBUG
     if (chunk != nullptr) {
         chunk->check_or_die();
-        CHECK(!chunk->is_empty());
```

Comment:
Why remove this?

Reply:
Look at the codes below, If `rows_to_keep > 0`, then we need to get more data from `IN_LAST_RESULT` and `Segment`(the previously processed part), and the `IN_LAST_RESULT` may be empty, but we need to generate empty sort columns through `ExprContext::evaluate` in method `_merge_sort_common`, this is why I remove the `DCHECK`

```cpp
    // Seoncd, there are two cases:
    // case1: rows_to_keep == 0, which measn part1 itself suffice, we don't need to check part2 and part3
    // case2: rows_to_keep > 0, which means part1 itself not suffice, we need to get more elements from part2 and part3,
    // so we need to first merge part2 and part3 and then get elements
    if (rows_to_keep > 0) {
        if (big_chunk == nullptr) {
            big_chunk.reset(segments[new_permutation.second[0].chunk_index].chunk->clone_empty(rows_to_keep).release());
        }
        const size_t sorted_size = _merged_segment.chunk->num_rows();
        rows_to_keep = std::min(rows_to_keep, sorted_size + second_size);
        if (_topn_type == TTopNType::RANK && sorted_size + second_size > rows_to_keep) {
            // For rank type, there may exist a wide equal range, so we need to keep all elements of part2 and part3
            rows_to_keep = sorted_size + second_size;
        }
        RETURN_IF_ERROR(_merge_sort_common(big_chunk, segments, rows_to_keep, sorted_size, new_permutation.second));
    }
```

Reply:
shouldn't pass the empty chunk to expr evaluate method, some exprs will crash